Libraries/LibWeb/Painting/BorderPainting.cpp

```diff
@@ -441,7 +444,10 @@ void paint_border(DisplayListRecorder& painter, BorderEdge edge, DevicePixelRect
             points.append(Gfx::FloatPoint(rect.bottom_left().to_type<int>()) + inner_corner);
         } else {
             Gfx::FloatPoint inner_right_angle_offset = Gfx::FloatPoint(0, opposite_joined_border_width.value() - opposite_radius.horizontal_radius);
-            points.append(Gfx::FloatPoint(rect.bottom_left().to_type<int>()) - inner_right_angle_offset);
+            auto bottom_left_point = Gfx::FloatPoint(rect.bottom_left().to_type<int>()) - inner_right_angle_offset;
+            // Clamp the final point to stay within the rect bounds to prevent rendering artifacts
+            bottom_left_point.set_y(AK::clamp(bottom_left_point.y(), rect.top().value(), rect.bottom().value()));
+            points.append(bottom_left_point);
```

Comment:
Instead of setting up a separate variable, you could just do `.constrained(rect)` to get the same result.